web/wp-content/plugins/tmbi-adobe-dtm/inc/adobe_dtm_utils.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/*
+ * User: jeysaravana
+ * Date: 2018-01-29
+ * Time: 12:18 PM
+ */
+
+class Adobe_DTM_Utils {
+	const PUBLISH_META_NAME = 'original_publish_date';
+
+	public $registered_post_type = array(
+		'post',
+		'page',
+		'listicle',
+		'collection',
+		'nicestplace',
+		'quiz',
+		'video',
+		'joke',
+		'marquee',
+		'project',
+		'tip',
+	);
+
+	public function __construct() {
+		add_action( 'admin_init', array( $this, 'adtm_init' ) );
+	}
+
+	public function adtm_init() {
+		//add_action( 'transition_post_status', array( $this, 'post_status_transition' ), 10, 3 );
+		//get_post_types()
+		add_action( 'publish_to_draft', array( $this, 'move_publish_to_draft' ) );
+		foreach ( $this->registered_post_type as $post_type ) {
+			add_action( 'publish_' . $post_type, array( $this, 'publishing_post' ), 10, 2 );
+		}
+	}
+
+	/*
+	 * Updating the meta tag when publishing the post. it will insert 'self::PUBLISH_META_NAME' at once
+	 *
+	 * @param int $post_id
+	 * @param WP_Post $post post object
+	 */
+	public function publishing_post( $post_id, $post ) {
+		if ( in_array( $post->post_type, $this->registered_post_type ) && ! get_post_meta( $post_id, self::PUBLISH_META_NAME, true ) ) {
+			update_post_meta( $post->ID, self::PUBLISH_META_NAME, $post->post_date );
+		}
+	}
+
+	/*
+	 * Updating the meta tag when change the post form publish to draft. it will insert 'self::PUBLISH_META_NAME' at once
+	 *
+	 * @param int $post_id
+	 * @param WP_Post $post post object
+	 */
+	public function move_publish_to_draft( $post ) {
+		if (  in_array( $post->post_type, $this->registered_post_type )
```

Comment:
There's a duplicate check here, but besides that code seems the same as in `publishing_post`. Hooks have different signatures so it may not be possible to unify them, but moving the duplicate code to a separate function (called from the two hooked actions) may reduce code duplication and help with maintainability.